temporian/core/serialization.py

```diff
@@ -55,95 +56,114 @@
         raise ValueError(
             "Can only save a function that has been compiled with"
             " `@tp.compile`."
         )
 
-    if not isinstance(inputs, dict):
-        raise ValueError(
-            "`inputs` must be a dictionary mapping the function's parameter"
-            " names to their corresponding values."
-        )
-
-    node_inputs = {}
-    inputs = inputs.copy()
-    for k, v in inputs.items():
-        node_input = _process_fn_input(v)
-        if node_input is not None:
-            inputs[k] = node_input
-            node_inputs[k] = node_input
+    merged_kwargs = _construct_kwargs_from_args_and_kwargs(
+        list(inspect.signature(fn).parameters.keys()), args, kwargs
+    )
+    node_kwargs = {k: _process_fn_input(v) for k, v in merged_kwargs.items()}
 
     # TODO: extensively check that returned types are the expected ones
-    outputs = fn(**inputs)
+    outputs = fn(**node_kwargs)
 
-    if isinstance(outputs, EventSetNode):
-        outputs = {"output": outputs}
+    outputs = _process_fn_outputs(outputs)
+
+    save_graph(inputs=node_kwargs, outputs=outputs, path=path)
 
-    if isinstance(outputs, dict):
-        if not all(isinstance(v, EventSetNode) for v in outputs.values()):
-            raise ValueError(
-                "The function must return a single EventSet or EventSetNode or"
-                " adictionary mapping output names to EventSets or"
-                " EventSetNodes."
-            )
 
-    save_graph(inputs=node_inputs, outputs=outputs, path=path)
+def load(
+    path: str,
+) -> Callable[..., Dict[str, EventSetNode]]:
+    """Loads a compiled Temporian function from a file.
 
+    The loaded function receives the same arguments and applies the same
+    operator graph as when it was saved.
 
-def _process_fn_input(input: Any) -> Optional[EventSetNode]:
-    # TODO: allow dicts and lists of stuff other than EventSet/EventSetNodes
-    if isinstance(input, (list, dict, tuple)):
-        raise ValueError(
-            "`tp.save` does not support collections in a function's input."
+    Args:
+        path: The path to load the function from.
+
+    Returns:
+        The loaded function.
+    """
+    with open(path, "rb") as f:
+        proto = text_format.Parse(f.read(), pb.Graph())
+
+    g: graph.Graph = _unserialize(proto)
+
+    assert g.named_inputs is not None
+    assert g.named_outputs is not None
+
+    @compile
+    def fn(
+        *args: EventSetNode,
+        **kwargs: EventSetNode,
+    ) -> Dict[str, EventSetNode]:
+        kwargs = _construct_kwargs_from_args_and_kwargs(
+            list(g.named_inputs.keys()), args, kwargs
         )
-    if isinstance(input, Schema):
-        return create_node_with_new_reference(schema=input)
-    if isinstance(input, EventSet):
-        return input.node()
-    if isinstance(input, EventSetNode):
-        return input
+        return g.apply_on_inputs(named_inputs=kwargs)

```

Comment:
This is what append when we mix functional and graph code :)

Nit: Can you move the "list(g.named_inputs.keys())" outside of the "fn" ?